src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,100 @@
+/*
+* This class parses a string to retrive tokens
+* and determines whether they are surrounded by qoutations or not
+*
+* author: Isaac Martins
+* version: May 19, 2017
+*/
+
+public final class Tokenizer
+{
+  //member variables
+  private StringBuilder token;
+  private String source;
+  private int at;
+
+  //functions (methods)
+  public Tokenizer(String source)
+  {
+    this.source = source;
+    at = 0;
+    token = new StringBuilder();
+  }
+
+  public String next() throws IOException
+  {
+    //Skip preceding whitespace
+    while(remaining() > 0 && Character.isWhitespace(peek()))
+    {
+      //ignore result because it's whitespace
+      read();
+    }
+    if(remaining() >= 0)
```

Comment:
Should this be `>=` or `<=`?

Part of the problem is that I'm not sure what this function should return. So you may want to use Javadoc comments on your functions (especially public ones like this) including bits like `@param` and `@return` to explain what inputs and outputs your function expects.

As it is, it sort of looks to me like the function will bail out with a null instead of the next token, because it will skip the preceding whitespace (good!), determine that there are characters left in the string via `remaining()`, and then bail. Is that right? If no, then what am I missing? If yes, then did unit tests reveal the problem, or do we need another test that will fail when there is a bug here?